internal/state/db/statepsql/channels.go

```diff
@@ -121,7 +122,7 @@ FROM
 
 	var cs []RawJSON
 	err := db.sql.SelectContext(ctx, &cs, q)
-	if err != nil {
+	if err != nil && err != sql.ErrNoRows {
 		return nil, xerrors.Errorf("exec select: %w")
```

Comment:
The reason it's not 404ing is because I forgot to wrap some of these errors. We don't want to ignore `sql.ErrorNoRows` because it'll just transparently return nil slices.